### PR TITLE
Add code examples for all public RemoteStorage functions

### PR DESCRIPTION
**Pull request by gregkare**
I have also indicated functions that take optional arguments (`authorize` and `connect`)

### CODE REVIEW

File: doc/js-api/remotestorage.rst [hunk 1]
```diff
@@ -152,11 +200,29 @@ The following functions can be called on your ``remoteStorage`` instance:
...
+  Example::
+
+     remoteStorage.stopSync();
+
 .. autofunction:: RemoteStorage#onChange(path, handler)
   :short-name:
+
+  Example::
+
+     remoteStorage.onChange('/bookmarks/', function() {
```
Comment (skddc) [inline doc/js-api/remotestorage.rst]: This is fired for directories? So it's different than the BaseClient change events?

Comment (gregkare) [reply doc/js-api/remotestorage.rst]: Let me check

Comment (gregkare) [reply doc/js-api/remotestorage.rst]: Yes, a directory as a path works, I tried it in myfavoritedrinks. That's what I understood from the current docs. It also works with a path to a document. Should I document both cases? I had only tried with a directory before

Comment (skddc) [reply doc/js-api/remotestorage.rst]: Oh wait, I misread this. Directories are the normal use case here. But it needs to link to the extensive docs in BaseClient.

Comment (gregkare) [reply doc/js-api/remotestorage.rst]: Ack! Adding it

Comment (galfert) [reply doc/js-api/remotestorage.rst]: You might also want to document that the handler function gets passed the change object in the example.

**Comment by skddc** [comment]
👍

**Pull request merged**